src/hoarder/archives/rar_archive_7z.py

```
@@ -0,0 +1,262 @@
+                raise ValueError(
+                    f"Directory {full_path} contains multiple non-indexed RAR files"
+                )
+            _, (scheme, rar_volumes) = rar_dict.popitem()
+            n_volumes = len(rar_volumes)
+            main_volume = rar_volumes[0]
+            try:
+                main_volume_path = main_volume.relative_to(storage_path)
+            except ValueError:
+                raise ValueError(
+                    f"Main volume {main_volume} is not under storage_path {storage_path}"
+                )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)